hpx/runtime/threads/policies/queue_holder_thread.hpp

```diff
@@ -210,15 +210,17 @@ namespace hpx { namespace threads { namespace policies {
           , queue_index_(queue)
           , thread_num_(thread_num)
           , owner_mask_(owner)
-          , min_delete_count_(init.min_delete_count_)
-          , max_delete_count_(init.max_delete_count_)
-          , max_terminated_threads_(init.max_terminated_threads_)
+          , min_delete_count_(static_cast<int>(init.min_delete_count_))
+          , max_delete_count_(static_cast<int>(init.max_delete_count_))
+          , max_terminated_threads_(
+                static_cast<int>(init.max_terminated_threads_))
           , terminated_items_(max_thread_count)
         {
             rollover_counters_.data_ =
                 std::make_tuple(queue_index_, round_robin_rollover);
             tq_deb.debug(debug::str<>("construct"), "D",
-                debug::dec<2>(domain_index_), "Q", debug::dec<3>(queue_index_),
+                debug::dec<2>(static_cast<int>(domain_index_)), "Q",
```

Comment:
The better way to do this would be 
`debug::dec<2,unsigned>(domain_index_)`
instead of
`debug::dec<2>(static_cast<int>(domain_index_))`
if the compiler is generating tons of warnings. (it would be nice if the default `int` could be removed and let the compiler deduce the right type)

Reply:
@biddisco Sure, please feel free to adjust the PR the way you feel is right.

Reply:
@biddisco I have changed the implementation of `dec` so it deduces the correct type now. In the future I'd like to map your 'logging' facilities onto the existing logging infrastructure, though.